src/internal/connector/exchange/service_iterators_test.go

```diff
@@ -0,0 +1,648 @@
+				displayName: "new",
+				p:           path.Builder{}.Append("2", "new"),
+			}),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "1", "deleted").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.DeletedState, false},
+				"2": {data.NewState, false},
+			},
+		},
+		{
+			name: "bad previous path strings",
+			getter: map[string]mockGetterResults{
+				"1": commonResults,
+			},
+			resolver: newMockResolver(mockResolvedContainer{
+				id:          "1",
+				displayName: "not_moved",
+				p:           path.Builder{}.Append("1", "not_moved"),
+			}),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  "1/fnords/mc/smarfs",
+				},
+				"2": DeltaPath{
+					delta: "old_delta_url",
+					path:  "2/fnords/mc/smarfs",
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.NewState, false},
+			},
+		},
+		{
+			name: "delta expiration",
+			getter: map[string]mockGetterResults{
+				"1": expiredResults,
+			},
+			resolver: newMockResolver(mockResolvedContainer{
+				id:          "1",
+				displayName: "same",
+				p:           path.Builder{}.Append("1", "same"),
+			}),
+			dps: DeltaPaths{
+				"1": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "1", "same").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.NotMovedState, true},
+			},
+		},
+		{
+			name: "a little bit of everything",
+			getter: map[string]mockGetterResults{
+				"1": commonResults,  // new
+				"2": commonResults,  // notMoved
+				"3": commonResults,  // moved
+				"4": expiredResults, // moved
+				// "5" gets deleted
+			},
+			resolver: newMockResolver(
+				mockResolvedContainer{
+					id:          "1",
+					displayName: "new",
+					p:           path.Builder{}.Append("1", "new"),
+				},
+				mockResolvedContainer{
+					id:          "2",
+					displayName: "not_moved",
+					p:           path.Builder{}.Append("2", "not_moved"),
+				},
+				mockResolvedContainer{
+					id:          "3",
+					displayName: "moved",
+					p:           path.Builder{}.Append("3", "moved"),
+				},
+				mockResolvedContainer{
+					id:          "4",
+					displayName: "moved",
+					p:           path.Builder{}.Append("4", "moved"),
+				},
+			),
+			dps: DeltaPaths{
+				"2": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "2", "not_moved").String(),
+				},
+				"3": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "3", "prev").String(),
+				},
+				"4": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "4", "prev").String(),
+				},
+				"5": DeltaPath{
+					delta: "old_delta_url",
+					path:  prevPath(suite.T(), "5", "deleted").String(),
+				},
+			},
+			expect: map[string]endState{
+				"1": {data.NewState, false},
+				"2": {data.NotMovedState, false},
+				"3": {data.MovedState, false},
+				"4": {data.MovedState, true},
```

Comment:
shouldn't this be false unless the token expired?

Reply:
note line 546 - `"4": expiredResults`